openrouteservice-api-tests/src/test/java/heigit/ors/services/matrix/ResultsValidationTest.java

```diff
@@ -89,95 +122,139 @@ public void emptyLocationEntryTest() {
+    }
+
+    @Test
+    public void resolveNamesParameterTest() {
+        Response response = given()
+                .param("locations", getParameter("locations"))
+                .param("sources", getParameter("sources1"))
+                .param("destinations", getParameter("destinations1"))
+                .param("profile", "driving-car")
+                .param("resolve_locations", "true")
+                .when()
+                .get(getEndPointName());
+
+        Assert.assertEquals(response.getStatusCode(), 200);
+        JSONObject jResponse = new JSONObject(response.body().asString());
+        Assert.assertEquals(true, jResponse.getJSONArray("sources").getJSONObject(0).has("name"));
+    }
+
+    private void checkTableDimensions(JSONObject json, String tableName, int rows, int columns) {
+        Assert.assertEquals(true, json.has(tableName));
+
+        JSONArray jTable = json.getJSONArray(tableName);
+        Assert.assertEquals(jTable.length(), rows);
+        Assert.assertEquals(jTable.getJSONArray(0).length(), columns);
+    }
+
+    @Test
+    public void distanceTest() {
+        Response response = given()
+                .param("locations", getParameter("manyLocations"))
+                .param("sources", "all")
+                .param("destinations","all")
+                .param("metrics", "distance")
+                .param("profile", "driving-car")
+                .param("resolve_locations", "true")
+                .when()
+                .get(getEndPointName());
+
+        String[] locations = (String[]) getParameter("manyLocationsArray");
+
+
+        Assert.assertEquals(response.getStatusCode(), 200);
+        JSONObject jResponse = new JSONObject(response.body().asString());
+        JSONArray jDistances = jResponse.getJSONArray("distances");
+        for(int i = 0; i < 12; i++) {
+            for (int j = 0; j < 12; j++) {
+                Response response2 = given()
+                        .param("coordinates", locations[i] + "|" + locations[j])
+                        .param("instructions", "false")
+                        .param("preference", getParameter("preference"))
+                        .param("profile", getParameter("carProfile"))
+                        .when()
+                        .get("routes");
+
+                JSONObject jResponseRouting = new JSONObject(response2.body().asString());
+                JSONObject jRoute = (jResponseRouting.getJSONArray("routes")).getJSONObject(0);
+                double routeDistance = jRoute.getJSONObject("summary").getDouble("distance");
+//                double matrixDistance;
+
+                double matrixDistance = jDistances.getJSONArray(i).getDouble(j);
+                Assert.assertTrue( matrixDistance - .1 < routeDistance);
+                Assert.assertTrue( matrixDistance + .1 > routeDistance);
+//                        .assertThat()
```

Comment:
We should try and make sure that we don't leave commented out code when we push to dev/master